reconcile/gitlab_housekeeping.py

```diff
@@ -156,20 +156,14 @@ def get_timed_out_pipelines(
 
 def clean_pipelines(
     dry_run: bool,
-    gl_instance: str,
-    gl_project_id: int,
-    gl_settings: str,
+    gl: GitLabApi,
+    fork_project_id: int,
     pipelines: list[dict],
 ) -> None:
-    if not dry_run:
-        with GitLabApi(
-            gl_instance, project_id=gl_project_id, settings=gl_settings
-        ) as gl:
-            gl_piplelines = gl.project.pipelines
-
     for p in pipelines:
         logging.info(["canceling", p["web_url"]])
         if not dry_run:
+            gl_piplelines = gl.get_project_by_id(fork_project_id).pipelines
```

Comment:
get project outside of `pipelines` for loop to save gitlab calls. btw can we directly pass `pipeline` object in this function and call `p.cancel()`? Or it's unsupported in this lib?

Reply:
We use GitLabApi.get_items() to get pipelines info, so it is a dict result. If we change to return typed object as it mentioned in TO DO, we can call p.cancel() directly.